lib/sequence.js

```diff
@@ -39,26 +38,41 @@ Sequence.prototype._run = function (fromIndex) {
 
 		var result = this.tests[i].run();
 
+		// if a Promise returned, we don't need to check for Promises after this test
+		// so we can just use Promise.each() on the rest of the tests
```

Comment:
@jamestalmage insisted on not creating Promises when they're not needed, because it slows down test suites with thousands of tests. I think we need to bring this discussion up again, as the code looks like (excuse me) shit, when we need to return variable async/sync result from Sequence/Concurrent#run().